From 042152d7c74fce0b8c803b3a02edfe0fcf26d6cf Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Fri, 6 May 2022 16:50:49 -0700
Subject: tf: allow CSV inputs to skip workloads (#38780)

Otherwise something like `--istio.test.skipWorkloads=vm,tproxy,delta`
will not work which leads to obscure errors
---
 pkg/test/framework/resource/flags.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pkg/test/framework/resource/flags.go b/pkg/test/framework/resource/flags.go
index 68162bab84..19161c0672 100644
--- a/pkg/test/framework/resource/flags.go
+++ b/pkg/test/framework/resource/flags.go
@@ -18,6 +18,7 @@
 	"flag"
 	"fmt"
 	"os"
+	"strings"
 
 	"istio.io/istio/pkg/test/env"
 	"istio.io/istio/pkg/test/framework/config"
@@ -58,6 +59,12 @@ func SettingsFromCommandLine(testID string) (*Settings, error) {
 		// TODO we may also want to trigger this if we have an old verion
 		s.SkipWorkloadClasses = append(s.SkipWorkloadClasses, "delta")
 	}
+	// Allow passing a single CSV flag as well
+	normalized := make(ArrayFlags, 0)
+	for _, sk := range s.SkipWorkloadClasses {
+		normalized = append(normalized, strings.Split(sk, ",")...)
+	}
+	s.SkipWorkloadClasses = normalized
 
 	if s.Image.Hub == "" {
 		s.Image.Hub = env.HUB.ValueOrDefault("gcr.io/istio-testing")
-- 
2.35.3

